modules/util/worker_handler.js

```diff
@@ -0,0 +1,21 @@
+import { createWorker } from './walkie_talkie';
+
+var workerInstance = createWorker('dist/iD-worker.js');
+
+export function parseBboxEntities(url, done) {
+    workerInstance.sendMessage(
+        { type: 'PARSE_BBOX_ENTITIES', url: url },
+        function(error, reply) {
+            done(error, reply);
+        }
+    );
+}
+
+export function authdParseBboxEntities(xhrParams, done) {
```

Comment:
I don't understand why there are separate auth'd and non auth'd parsers.  At the point that the worker gets involved haven't they already been downloaded?

Reply:
The worker code makes the network request to OSM. There are two possible network requests that iD wants to make, one with authentication headers and the other without it. Since the worker makes the network request and not the main iD, I made two separate worker functions to do the same.

Reply:
Yea but I thought you said you were going to start off with a version that doesn't do the request, and only does the tile parsing, since that's easier.

Reply:
Ya, I did intend to start off with that but then the passing back and forth between worker and main is too slow. I think letting the Webworker make the network request is the best possible way we can offload the entire task and with the current approach I think it is working quite fine.